pkg/query/remote_engine.go

```diff
@@ -245,7 +250,8 @@ func (r *remoteQuery) Exec(ctx context.Context) *promql.Result {
 		}
 
 		if warn := msg.GetWarnings(); warn != "" {
-			return &promql.Result{Err: errors.New(warn)}
+			warnings = append(warnings, errors.New(warn))
+			continue
```

Comment:
Actually I am wondering if things need to change since remote engine (can be any implementation) might return both warning and results the same time? Can we safely ignore the results?

Reply:
Agreed, it seems like either way we'd skip to the next iteration couple lines below if there are no series?

Reply:
I thought we can only have either a warning or a series in a response, at least that's how I understand the `oneOf` clause in the proto:
https://github.com/thanos-io/thanos/blob/221881b4f882973c389585208729c27aea7c1ec9/pkg/api/query/querypb/query.proto#L57-L65

Reply:
You're right, I didn't realize it's `oneof` field, then this makes sense, thanks 👍

Reply:
This is the proto def in Thanos, right? It is defined by us.
For the Prometheus interface itself, select can return both warning and results the same time?
I wonder if the oneof makes sense or not.